staging/src/k8s.io/component-base/zpages/statusz/statusz_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusz
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/blang/semver/v4"
+	"github.com/google/go-cmp/cmp"
+)
+
+const wantTmpl = `
+%s statusz
+Warning: This endpoint is not meant to be machine parseable, has no formatting compatibility guarantees and is for debugging purposes only.
+
+Started: %v
+Up: %s
+Go version: %s
+Binary version: %v
+Emulation version: %v
+Minimum Compatibility version: %v
+`
+
+var (
+	fakeStartTime               = time.Now()
+	fakeUptime                  = uptime(fakeStartTime)
+	fakeGoVersion               = "1.21"
+	fakeBinaryVersion           = semver.Version{Major: 1, Minor: 31, Patch: 0}
+	fakeEmulationVersion        = semver.Version{Major: 1, Minor: 30, Patch: 0}
+	fakeMinCompatibilityVersion = semver.Version{Major: 1, Minor: 29, Patch: 0}
+)
+
+func TestStatusz(t *testing.T) {
+	delimiters = []string{":"}
+	tests := []struct {
+		name           string
+		componentName  string
+		reqHeader      string
+		registry       fakeRegistry
+		wantStatusCode int
+		wantBody       string
+	}{
+		{
+			name:           "invalid header",
+			reqHeader:      "some header",
+			wantStatusCode: http.StatusNotAcceptable,
+		},
+		{
+			name:          "valid request",
+			componentName: "test-server",
+			reqHeader:     "text/plain; charset=utf-8",
```

Comment:
Test `*/plain`, `text/*`, and some invalid cases?

Reply:
Added more test cases